src/wp-includes/theme.php

```diff
@@ -3142,3 +3142,41 @@ function _wp_keep_alive_customize_changeset_dependent_auto_drafts( $new_status,
 		clean_post_cache( $post_id );
 	}
 }
+/**
+ * Get the theme logo through customizer.
+ * Add style with background image if theme logo exists. 
+ * Note: By default, the size of your logo should be no bigger than 80 x 80 pixels 
+ * (though even this can change with custom CSS). 
+ * Also change the link of the <a> link from the blog-url and title attribute from blog description
+ */
+function alter_login_logo() { 
+	$custom_logo_id = get_theme_mod( 'custom_logo' );
+	$image = wp_get_attachment_image_src( $custom_logo_id , 'full' );
+	$url_image = $image[0];
```

Comment:
This is probably going to cause a PHP notice, which we should avoid.

When there is no logo set, what does `$custom_logo_id` contain?

When it contains the ID of an invalid image (deleted, for example), what does `wp_get_attachment_image_src` return?

We should check for each of these conditions and each one should have its own `return` statement.  
Something like this may be sufficient (note, untested):

```php
$custom_logo_id = get_theme_mod( 'custom_logo' );
if ( empty( $custom_logo_id ) ) {
    return;
}
$image = wp_get_attachment_image_src( $custom_logo_id, 'full' );
if ( ! is_array( $image ) || ! isset( $image[0] ) ) {
    return;
}
$url_image = $image[0];
```

Also, this bit of logic should be broken out into a new function like `get_logo_image_url()` and each of these 3 cases (logo exists, logo points to an invalid ID, no logo is set) should be covered by an automated test.